### ISSUE TITLE
Making a carousel slideshow with the active slide in the center

### ISSUE THREAD

**Issue by rasgar** [malsup/cycle2#112]
Is there any way to do this? I want to have the active slide in the center of the page with the previous slide to the left and the next slide to the right, as opposed to having the active one on the far left.

**Comment by daltonrooney** [comment]
I've seen this done in the past by syncing three slideshows: http://rachelhulin.com/book1/?c=1-the-flying-series

**Comment by albell** [comment]
Why do you want to do this? If it's for styling, there might be a way to get what you want by chaining adjacent sibling selectors. E.g. to fade back the first slide and third slides but leave the second slide opaque:

<pre>
.cycle-slide-active {opacity: .5}
.cycle-slide-active + .cycle-slide {opacity: 1}
.cycle-slide-active + .cycle-slide + .cycle-slide {opacity: .5}
</pre>

That's pretty easy, but there will be a compatibility problem with IE 6, because it doesn't support CSS2.1. More robust (but more work to maintain) would be to try hacking line 457 of the source to add the "active" class to the nextSlide instead of the currSlide. Hope this helps...

**Comment by rasgar** [comment]
The main problem is that I have an overlay with a caption in the middle of the slideshow. Because the first slide is the "active" one, I have the caption from the left slide overlaying the middle one.

I can't do the three adjacent slideshows, as the one I'm working with is associated with a CMS and it dynamically generates a list of images from a database.

As for editing the source... I was able to get it to place the active class on the slide I want, so thanks for that tip. Unfortunately it does not display the correct caption when I do that. I am trying to modify the section for captions, but so far I can't figure it out.

**Comment by albell** [comment]
Cool, you're almost there. You just need to modify the caption plugin to template and update the nextSlide instead of the currSlide. Look at lines 681 and 688.

**Comment by brrrm** [comment]
Would it be an idea to hook into the slideEvent and calculate a different value for the CSS Left position for the slideshow-wrapper and apply that after sliding and after initializing?

**Comment by rasgar** [comment]
@baloneysandwiches For whatever reason my line numbers aren't quite matching up with yours, but I'm pretty sure I know what you're talking about. Is it this segment here?

$(document).on( 'cycle-update-view', function( e, opts, slideOpts, **currSlide** ) {
    if ( opts.captionModule !== 'caption' )
        return;
    var el;
    $.each(['caption','overlay'], function() {
        var name = this; 
        var template = slideOpts[name+'Template'];
        var el = opts.API.getComponent( name );
        if( el.length && template ) {
            el.html( opts.API.tmpl( template, slideOpts, opts, **currSlide** ) );

I've already changed the bolded currSlide to nextSlide, and it didn't change anything at all. I don't know what else to try.

@brrrm I'm not quite sure what that means. What is the slideEvent, and how would CSS positioning work for this?

**Comment by brrrm** [comment]
@rasgar see the API docs: http://jquery.malsup.com/cycle2/api/#events
You can create an eventListener that fires after cycling: $('#my-slideshow').on('cycle-after', function(){ // do something });

If you re-position the cycle-wrapper with absolute positioning so it will show half of the previous and half of the next slide I think you can achieve what you want.

**Comment by rasgar** [comment]
I'm not quite sure if I understand what you're getting at. For one thing, I can't figure out what exactly I need to write to use that function to modify the left positioning. That's probably my own fault, though, for not being very familiar with JavaScript stuff.
But the other thing is, how will this differ from simply placing a negative margin on the div that I placed the slideshow in?

**Comment by albell** [comment]
@rasgar Actually, there is a pure css solution. Just add a left margin equal to your image width to the cycle-carousel-wrap element. I just tried this plugging it into Dev Tools on the "controls" demo page:

http://www.malsup.com/jquery/cycle2/demo/carousel.php

and it works fine. I even tried completing the loop, going all the way through by clicking on the "next" button, and there's never any glitch or unwanted offset.

Put it in your css with an !important declaration just to be sure it doesn't ever get overwritten by the plugin:

<pre>
.cycle-carousel-wrap { margin-left: 100px !important} (or whatever your slide width is)
</pre>

and go back to using the unaltered source. Much cleaner!

Now if your slides are all different widths, or if you need fully responsive behavior down to mobile sizes, then things have the potential to get tricky, and you will need to get into some media queries at the very least. (P.S. if you're just starting out with jquery I recommend the sitepoint book jquery: novice to ninja. It put me on the the path. No, i'm not a Sitepoint shareholder.)

**Comment by rasgar** [comment]
@baloneysandwiches Thank you so much, this solution is absolutely perfect! Due to the way the slideshows work on all of our clients' sites, I am going to be able to use this for every site I work on. All the slides will always be the same size no matter what.

**Comment by albell** [comment]
@rasgar Great. One last thought: If it's always going to be three slides side-by-side, _and_ your container has an explicit css width (easy to do), you could preserve responsiveness by using a percentage for the margin:

<pre>
.cycle-carousel-wrap { margin-left: 33.33% !important}
</pre>

**Comment by malsup** [comment]
This is also what the <code>data-cycle-carousel-offset</code> option does (for pixel-sized slideshows).

**Comment by hasft** [comment]
is there any another solution for this like 
.cycle-carousel-wrap { margin: 0 auto !important}
?

example:
http://www.wistar.org/

**Comment by Gef** [comment]
Here is some CSS to get a full width centred slideshow using carousel like the wistar example above.

.slideshow-outer {
  width: 100%;
  overflow: hidden;
  position: relative;
}
.slideshow-wrapper {
  left: 50% !important;
  position: relative;
  margin-left: -480px !important;
  overflow: visible !important;
  margin: 0 auto;
  height: 300px;
  width: 960px;
}
.cycle-slideshow {
  height: 300px;
  left: -100%;
}
.cycle-carousel-wrap {
    margin-left: 33.33333%!important;
 }

**Comment by natezander** [comment]
I have the need for this technique for a carousel below my main slideshow - with the center item of the carousel pointing to the slideshow.

I have tried malsup's technique of data-cycle-carousel-offset, but when I do that, my paging carousel can't reach the final two slides (because they're offset). I used a % value for the offset.

The CSS solution by Gef presented the same issue. Am I missing something?

Current Design:
![screen shot 2013-11-20 at 11 36 48 am](https://f.cloud.github.com/assets/5290498/1584830/dca45d1e-5212-11e3-87f5-734ea0d29f73.png)

Desired Design:
![screen shot 2013-11-20 at 11 37 17 am](https://f.cloud.github.com/assets/5290498/1584834/e44c15e8-5212-11e3-8643-eaa905b90162.png)
